livekit-agents/livekit/agents/utils/pool.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import time
+from typing import Awaitable, Callable, Generic, Optional, Set, TypeVar
+
+T = TypeVar("T")
+
+logger = logging.getLogger("livekit.agents.pool")
+
+
+class ConnectionPool(Generic[T]):
```

Comment:
I think lot of the exposed API could be made private? Like connect()

Reply:
renamed connect. what else do you see that should be private?